File: Kotlin_5/FunctionWithDefaultParameters.java
package Kotlin_5;

public class FunctionWithDefaultParameters {
    // 기본 매개변수를 사용하는 것과 유사하게, 일부 인자만으로도 객체를 생성할 수 있음.
    Person3 p1 = new Person3("KangSungJun");
    Person3 p2 = new Person3("KangSungJun", "SomeWhere");

    // 모든 인자를 다 입력할 수도 있음.
    Person3 p3 = new Person3("KangSungJun", "SomeWhere", true);

    // 기본 매개변수를 사용하는 것과 유사하게, 일부 인자만으로도 메서드를 호출할 수 있음.
}

File: Kotlin_5/FunctionWithDefaultParameters.kt
package Kotlin_5

// 기본 매개변수가 있는 주 생성자에 @JvmOverloads 어노테이션을 적용함.
class Person3 @JvmOverloads constructor(val name : String, var address : String = "", var isAdult : Boolean = false){
    // 기본 매개변수가 있는 함수에 @JvmOverloads 어노테이션을 적용함.
    @JvmOverloads
    fun doSomething(a : String, b : Int = 0, c : Boolean = false){
    }
}